pkg/manager/member/ticdc_member_manager.go

```diff
@@ -0,0 +1,312 @@
+		err = controller.SetServiceLastAppliedConfigAnnotation(&svc)
+		if err != nil {
+			return err
+		}
+		_, err = tcmm.svcControl.UpdateService(tc, &svc)
+		return err
+	}
+
+	return nil
+}
+
+func getNewCDCHeadlessService(tc *v1alpha1.TidbCluster) *corev1.Service {
+	ns := tc.Namespace
+	tcName := tc.Name
+	instanceName := tc.GetInstanceName()
+	svcName := controller.TiCDCPeerMemberName(tcName)
+	svcLabel := label.New().Instance(instanceName).TiCDC().Labels()
+
+	svc := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            svcName,
+			Namespace:       ns,
+			Labels:          svcLabel,
+			OwnerReferences: []metav1.OwnerReference{controller.GetOwnerRef(tc)},
+		},
+		Spec: corev1.ServiceSpec{
+			ClusterIP: "None",
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "ticdc",
+					Port:       8301,
+					TargetPort: intstr.FromInt(int(8301)),
+					Protocol:   corev1.ProtocolTCP,
+				},
+			},
+			Selector:                 svcLabel,
+			PublishNotReadyAddresses: true,
+		},
+	}
+	return &svc
+}
+
+func getNewTiCDCStatefulSet(tc *v1alpha1.TidbCluster) (*apps.StatefulSet, error) {
+	ns := tc.GetNamespace()
+	tcName := tc.GetName()
+
+	baseTiCDCSpec := tc.BaseTiCDCSpec()
+	ticdcLabel := labelTiCDC(tc)
+	stsName := controller.TiCDCMemberName(tcName)
+	podAnnotations := CombineAnnotations(controller.AnnProm(8301), baseTiCDCSpec.Annotations())
+	stsAnnotations := getStsAnnotations(tc, label.TiCDCLabelVal)
+	headlessSvcName := controller.TiCDCPeerMemberName(tcName)
+
+	cmdArgs := []string{"/cdc server", "--addr=0.0.0.0:8301", "--advertise-addr=${POD_NAME}.${HEADLESS_SERVICE_NAME}.${NAMESPACE}.svc:8301"}
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--pd=http://%s-pd:2379", tcName))
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--tz=%s", tc.TiCDCTimezone()))
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--gc-ttl=%d", tc.TiCDCGCTTL()))
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--log-file=%s", tc.TiCDCLogFile()))
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--log-level=%s", tc.TiCDCLogLevel()))
+	cmd := strings.Join(cmdArgs, " ")
+
+	env := []corev1.EnvVar{
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+		{
+			Name: "NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name:  "HEADLESS_SERVICE_NAME",
+			Value: headlessSvcName,
+		},
+	}
+
+	ticdcContainer := corev1.Container{
+		Name:            v1alpha1.TiCDCMemberType.String(),
+		Image:           tc.TiCDCImage(),
+		ImagePullPolicy: baseTiCDCSpec.ImagePullPolicy(),
+		Command:         []string{"/bin/sh", "-c", cmd},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "ticdc",
+				ContainerPort: int32(8301),
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Resources: controller.ContainerResource(tc.Spec.TiCDC.ResourceRequirements),
+		Env:       env,
```

Comment:
Handle the env like [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tidb_member_manager.go#L702)